### PR TITLE
chore: add prettier config

### PR DESCRIPTION
**Pull request by WinterYukky**
Add Prettier as a code styling tool.
By introducing prettier, it is possible to make the code style consistent. Also, if there are contributors using prettier as an IDE, their code will be formatted based on prettier's default settings, but they may conflict with Eslint. It would be a good idea to prepare settings on the project side.

Fortunately, projen supports prettier, so you only need to set up a few lines of `projenrc.ts`.

### CODE REVIEW

File: src/aws-cur/cost-report.ts [hunk 1]
```diff
@@ -5,13 +5,18 @@ import { Construct } from 'constructs';
  * Enum for the possible granularities of a cost report
  */
 export class ReportGranularity {
-
   /** Hourly granularity */
-  public static readonly HOURLY: ReportGranularity = new ReportGranularity('HOURLY');
+  public static readonly HOURLY: ReportGranularity = new ReportGranularity(
```
Comment (hoegertn) [inline src/aws-cur/cost-report.ts:9]: I am not a fan of this change

File: src/aws-ec2/instance-connect-endpoint.ts [hunk 2]
```diff
@@ -4,7 +4,9 @@ import { Construct } from 'constructs';
 /**
  * An EC2 Instance Connect Endpoint.
  */
-export interface IInstanceConnectEndpoint extends aws_ec2.IConnectable, IResource {
+export interface IInstanceConnectEndpoint
```
Comment (hoegertn) [inline src/aws-ec2/instance-connect-endpoint.ts:7]: Please not

File: src/aws-ec2/instance-connect-endpoint.ts [hunk 3]
```diff
@@ -130,7 +139,7 @@ export class InstanceConnectEndpoint extends Resource implements IInstanceConnec
     return new aws_ec2.CfnInstanceConnectEndpoint(this, 'Resource', {
       clientToken: this.props.clientToken,
       preserveClientIp: this.props.preserveClientIp,
-      securityGroupIds: this.securityGroups.map(sg => sg.securityGroupId),
+      securityGroupIds: this.securityGroups.map((sg) => sg.securityGroupId),
```
Comment (hoegertn) [inline src/aws-ec2/instance-connect-endpoint.ts:142]: debatable

File: test/aws-ec2/instance-connect-endpoint.test.ts [hunk 4]
```diff
@@ -18,12 +18,20 @@ describe('InstanceConnectEndpoint', () => {
       }),
     });
 
-    Template.fromStack(stack).hasResourceProperties('AWS::EC2::InstanceConnectEndpoint', {
```
Comment (hoegertn) [inline test/aws-ec2/instance-connect-endpoint.test.ts:21]: In general yes, but sometimes I think it is better readable to collapse some KV-pairs

**Review by hoegertn — CHANGES_REQUESTED** [review]
I love the idea
I am not a fan of the format

**Comment by mrgrain** [comment]
This is certainly a good idea. 👍

Personally I don't have a strong opinion about specific formatting options as long as it's consistent and enforced.

**Comment by WinterYukky** [comment]
As long as it's consistent like @mrgrain , I'm not particular about anything else. I tried increasing the range allowed by Prettier so that it would be @hoegertn 's favorite format.

File: .projenrc.ts [hunk 5]
```diff
@@ -50,6 +65,18 @@ const project = new awscdk.AwsCdkConstructLibrary({
...
   ],
+  eslintOptions: {
+    dirs: ['src', 'test'],
+    prettier: true,
+  },
+  prettier: true,
+  prettierOptions: {
+    settings: {
+      singleQuote: true,
+      printWidth: 100,
```
Comment (mrgrain) [inline .projenrc.ts:76]: The previous ESLint rule used 150. I think that's maybe a bit excessive, but 120 might be a good compromise.

```suggestion
      printWidth: 120,
```

**Comment by WinterYukky** [comment]
Self-mutation is also a good match. If a contributor forgets to format it, there's no need to point it out.

**Comment by hoegertn** [comment]
Still not a big fan of the excessive newlines. Can't we use prettier but with the same settings as before?
These were a good fit and are also basically what every CDK / projen projects uses.

**Comment by dbartholomae** [comment]
Prettier, [by design](https://prettier.io/docs/en/option-philosophy), isn't really that configurable, so I'm not sure if we can match specific ideas. To be honest, I would personally prefer to use prettier with as little changes as possible, as this enforces the notion that codestyle usually isn't the most important thing to discuss - but I'm also in the camp that doesn't really care how exactly we format the code.

My suggestion would be maybe @hoegertn you can check [which of the options](https://prettier.io/docs/en/options) you care about and we can then update the config accordingly. I don't think we will be able to pixel-perfect match the old style, but maybe we can at least get close on what matters.

**Comment by hoegertn** [comment]
I think I am fine with the selected options but I really hate the excessive newlines.

But if we decide we want to use prettier I will live with it.

**Review by hoegertn — APPROVED** [review]
Not a fan but ok with it

**Pull request merged**